### PR TITLE
Lobby fixes

### PR DESCRIPTION
**Pull request by yummyblabla**
- [x] User goes to Game page when START_GAME listener is called
 - [x] Proper game list update when host leaves their created game
 - [x] Game Page and Game Test page created with added pixijs package
 - [x] Other players gets UI update that host leaves the game lobby

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,13 @@
   "name": "mahjong-kings",
   "version": "0.1.0",
   "private": true,
+  "jest": {
+    "collectCoverageFrom": [
+      "src/*.{js,jsx,ts,tsx}",
+      "!<rootDir>/node_modules/",
```
Comment (jzc1828) [inline package.json]: collect coverage from `node_modules`?

File: src/components/MainThreeJSComponent.tsx [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import React, { useEffect, useRef } from 'react';
 import * as THREE from 'three';
 
-import CHUN from '../assets/tiles/Regular/Chun.png';
-import MAN1 from '../assets/tiles/Regular/Man1.png';
-import HATSU from '../assets/tiles/Regular/Hatsu.png';
-import NAN from '../assets/tiles/Regular/Nan.png';
-import PEI from '../assets/tiles/Regular/Pei.png';
-import SOU1 from '../assets/tiles/Regular/Sou1.png';
+import REDDRAGON from '../assets/tiles/Regular/REDDRAGON.png';
```
Comment (jzc1828) [inline src/components/MainThreeJSComponent.tsx:4]: these changes are already in master, maybe ur master branch is not up to date? need to merge or rebase master into ur branch

File: src/components/lobby/game_lobby.tsx [hunk 3]
```diff
@@ -88,14 +91,32 @@ const GameLobby = ({ ws, game, gameRef, setGame }: GameLobbyProps): JSX.Element
     setGame(newGame);
   };
 
+  /**
+   * For LEAVE_GAME
+   * @param payload LeaveGameJSON
+   */
+  const leaveGame = (payload: unknown): void => {
+    const data = payload as LeaveGameJSON;
+    const { success, game: g, error } = data;
```
Comment (jzc1828) [inline src/components/lobby/game_lobby.tsx:100]: is `g` an alias? never use this before, so I'm just curious

File: package.json [hunk 4]
```diff
@@ -2,6 +2,12 @@
   "name": "mahjong-kings",
   "version": "0.1.0",
   "private": true,
+  "jest": {
+    "collectCoverageFrom": [
+      "src/modules/**.{js,jsx,ts,tsx}",
+      "!<rootDir>/node_modules/"
```
Comment (jzc1828) [inline package.json:8]: collect test coverage from `node_modules`?

File: src/components/lobby/game_lobby.tsx [hunk 5]
```diff
@@ -106,20 +127,31 @@ const GameLobby = ({ ws, game, gameRef, setGame }: GameLobbyProps): JSX.Element
...
   return (
-    <div>
-      <p>Game Lobby</p>
+    <div className="padding-30">
+      <p className="padding-10 color-white">Game Lobby</p>
+      {game && (
+        <>
+          <p>Game Info</p>
+          <p>{`Game Name: ${game.gameName}`}</p>
+          <p>{`Game Type: ${game.gameVersion}`}</p>
```
Comment (jzc1828) [inline src/components/lobby/game_lobby.tsx:150]: typo, should be `gameType` here?
```
<p>{`Game Type: ${game.gameType}`}</p>
```

File: src/modules/mahjong/Hand/Hand.ts [hunk 6]
```diff
@@ -20,9 +20,9 @@ interface SortHandWeights {
   [HonorTileTypes.SOUTH]: number;
   [HonorTileTypes.WEST]: number;
   [HonorTileTypes.NORTH]: number;
-  [HonorTileTypes.GREEN_DRAGON]: number;
-  [HonorTileTypes.RED_DRAGON]: number;
-  [HonorTileTypes.WHITE_DRAGON]: number;
+  [HonorTileTypes.GREENDRAGON]: number;
```
Comment (jzc1828) [inline src/modules/mahjong/Hand/Hand.ts:23]: same as above, need to rebase master

File: src/modules/mahjong/Hand/Hand.ts [hunk 7]
```diff
@@ -78,9 +78,9 @@ class Hand {
       [HonorTileTypes.SOUTH]: south,
       [HonorTileTypes.WEST]: west,
       [HonorTileTypes.NORTH]: north,
-      [HonorTileTypes.GREEN_DRAGON]: green,
-      [HonorTileTypes.RED_DRAGON]: red,
-      [HonorTileTypes.WHITE_DRAGON]: white,
+      [HonorTileTypes.GREENDRAGON]: green,
```
Comment (jzc1828) [inline src/modules/mahjong/Hand/Hand.ts:81]: same as above, need to rebase master

File: src/modules/mahjong/Wall/init/Honor.ts [hunk 8]
```diff
@@ -42,13 +42,13 @@ const honorTileInit: HonorTileInitObject = {
     type: HonorTileTypes.NORTH,
   },
   red: {
-    type: HonorTileTypes.RED_DRAGON,
+    type: HonorTileTypes.REDDRAGON,
```
Comment (jzc1828) [inline src/modules/mahjong/Wall/init/Honor.ts:45]: same as above, need to rebase master

File: src/pages/Lobby.tsx [hunk 9]
```diff
@@ -99,6 +97,19 @@ const LobbyPage = ({ ws, currentUser }: LobbyProps): JSX.Element => {
...
   };
 
+  /**
+   * Remove game from games list.
+   * @param gameId string
+   */
+  const removeGameFromList = (gameId: string) => {
+    const newGames = [...games];
+    const index = games.findIndex((game) => game.gameId === gameId);
+    if (index !== 1) {
```
Comment (jzc1828) [inline src/pages/Lobby.tsx:107]: do you mean not equal to `-1`? so u can find the game in the `games` array?
```
if (index !== -1) {
```

File: package.json (hunk 4 above)
Comment (yummyblabla) [reply package.json:8]: ! is not including

File: src/components/lobby/game_lobby.tsx (hunk 3 above)
Comment (yummyblabla) [reply src/components/lobby/game_lobby.tsx:100]: you can use aliases if there are variable name conflicts

File: src/pages/Lobby.tsx (hunk 9 above)
Comment (yummyblabla) [reply src/pages/Lobby.tsx:107]: Fixed

File: package.json (hunk 4 above)
Comment (jzc1828) [reply package.json:8]: kk

File: src/components/lobby/game_lobby.tsx (hunk 5 above)
Comment (jzc1828) [reply src/components/lobby/game_lobby.tsx:150]: need to fix this

Comment (yummyblabla) [reply src/components/lobby/game_lobby.tsx]: Done

**Pull request merged**